### DISCUSSION TITLE
Handling multiple contacts on ContactAdded (for e.g. to play contact sound)

### DISCUSSION THREAD

**Discussion by wrightwriter** [jrouwe/JoltPhysics#865] [Q&A] [answered]
I have an ConvexHull dynamic body (icosphere) colliding with a MeshShape static body(terrain). It seems to generate a bunch of ContactAdded events. If I, say, want to play a single sound on collision, what's a good way to handle it?

I was thinking of summing all forces from all contact collisions on that body, then using that as a heuristic for playing the sound. Does that make sense?

Here's a demonstration video. I'm calculating force (mContactImpulse/deltaTime of the largest impulse in the contact), then printing it if it's larger than a threshold (also scaling down by 100):

https://github.com/jrouwe/JoltPhysics/assets/8731857/2abd57a4-db5b-44c3-83e9-41e6e429a4a5

**Comment by jrouwe** [comment, marked as answer]
I think you indeed have to use some heuristic like summing all the contact impulses, but you probably want to take into account when you hit different bodies with different material types, so the summing should be somewhat selective. If you divide contact impulse by mass of the object then you get 'delta velocity' which may be a more representative number than force because it is more uniform between different bodies (note that this is crude, it will not be the actual delta velocity of the body as the contact is not at the center of mass and you're not taking the mass of the other object into account).

**Comment by wrightwriter** [comment]
Thank you, I summed delta velocity, works great! Hope this helps someone else as well.

https://github.com/jrouwe/JoltPhysics/assets/8731857/5ed3f361-9883-481f-a8f2-4b55c23e35b3

(btw not sure if videos count to repo size, can delete them)

**Reply by jrouwe** [reply]
Nice!

> btw not sure if videos count to repo size, can delete them

I don't think they do (and I don't think I can delete them either)
